keymanapp/keyman
# refactor(android): Change com.tavultesoft.kmea package to com.keyman.engine 🛋

> If possible, @mcdurdin would also like to move the KMManager package 
> from `com.tavultesoft.kmea` to `com.keyman.engine.manager`

_Originally posted by @darcywong00 in https://github.com/keymanapp/keyman/issues/7881#issuecomment-1371761542_
      
There were too many uses of "protected" members in the com.tavultesoft.kmea classes, so I wasn't able to isolate KMManager.java into `com.keyman.engine.manager`.
So the next best thing I could do was mass-rename the package from `com.tavultesoft.kmea` to `com.keyman.engine`.
This rippled to the package names in KMEA Android Manifest.xml

But I'm intentionally leaving apps like "Keyman for Android" at `com.tavultesoft.kmapro` since they're published.

The only change that wasn't a rename was for some reason FileUtilsTest changed the contents in the logs from 3 items to 2.

## Breaking change for KMEA consumers

Consumers of Keyman Engine for Android will need to update their references from `com.tavultesoft.kmea` to `com.keyman.engine` accordingly.

## User Testing
This PR affected practically every Android source file, so check that all the Android apps load and behave correctly
Setup - For each test, load the PR build of the app on an Android device/emulator. 

* **TEST_KEYMAN** - Verifies Keyman for Android
1. Install and launch the PR build of "Keyman for Android"
2. Verify app loads fine and functions normally

* **TEST_KMSAMPLE1** - Verifies KMSample1 app
1. Install and launch the Android PR build of KMSample1 
2. Verify app loads fine and functions normally

* **TEST_KMSAMPLE2** - Verifies KMSample2 app
1. Install and launch the Android PR build of KMSample2 
2. Verify app loads fine and functions normally

* **TEST_FV_ANDROID** - Verifies FirstVoices for Android app
1. Install and launch the Android PR build of "FirstVoices for Android"
2. Verify app loads fine 
3. From the setup menu, install and enable the keyboard for BC Coast --> Sencoten
4. Allow time for the dictionary to also install in the background
5. Complete the setup menu so FV for Android is enabled as a system keyboard
6. Launch a separate app (e.g. Chrome) and select a text field to type in
7. With FV Android as the selected keyboard, verify it functions normally
